src/wazuh_modules/wm_database.c

```diff
@@ -401,6 +407,20 @@ void wm_sync_agents() {
 
 #endif // LOCAL
 
+void wm_get_os_arch(char * os_header, char *os_arch) {
+    if      (strstr(os_header, "x86_64") != NULL) snprintf(os_arch, strlen("x86_64")+1, "%s", "x86_64");
+    else if (strstr(os_header, "i386") != NULL) snprintf(os_arch, strlen("i386")+1, "%s", "i386");
+    else if (strstr(os_header, "i686") != NULL) snprintf(os_arch, strlen("i686")+1, "%s", "i686");
+    else if (strstr(os_header, "sparc") != NULL) snprintf(os_arch, strlen("sparc")+1, "%s", "sparc");
+    else if (strstr(os_header, "amd64") != NULL) snprintf(os_arch, strlen("amd64")+1, "%s", "amd64");
+    else if (strstr(os_header, "ia64") != NULL) snprintf(os_arch, strlen("ia64")+1, "%s", "ia64");
+    else if (strstr(os_header, "AIX") != NULL) snprintf(os_arch, strlen("AIX")+1, "%s", "AIX");
+    else if (strstr(os_header, "armv6") != NULL) snprintf(os_arch, strlen("armv6")+1, "%s", "armv6");
+    else if (strstr(os_header, "armv7") != NULL) snprintf(os_arch, strlen("armv7")+1, "%s", "armv7");
+    else                                         memset(os_arch,'\0',sizeof(os_arch));
```

Comment:
`sizeof(array)` is only valid if the array is spilled in the same function. In this case it will return the size of the pointer (8) no matter the length of the array. Consider using the `size` variable.

Or if you choose to use `strdup()` you may call `strdup("")`, it will return a valid (non-null) empty string.